Cabal/src/Distribution/Simple/Utils.hs

```diff
@@ -389,6 +389,11 @@ dieWithException verbosity exception = do
   ts <- getPOSIXTime
   throwIO $ VerboseException callStack ts verbosity exception
 
+dieWithExceptionCabalInstall :: HasCallStack => Verbosity -> CabalInstallException -> IO a
+dieWithExceptionCabalInstall verbosity exception = do
+  ts <- getPOSIXTime
+  throwIO $ VerboseException callStack ts verbosity exception
```

Comment:
In fact, we don't need to introduce this function at all. We can simply replace its uses with `dieWithException` and generalize the type signature of that to be more polymorphic, I believe.